diff -urN tdegraphics.orig/config.h.cmake tdegraphics/config.h.cmake
--- tdegraphics.orig/config.h.cmake	2019-02-19 23:03:51.959567176 +0300
+++ tdegraphics/config.h.cmake	2019-02-19 23:05:02.916565869 +0300
@@ -1,6 +1,7 @@
 #cmakedefine VERSION "@VERSION@"
 
 // poppler-tqt
+#cmakedefine HAVE_POPPLER_064
 #cmakedefine HAVE_POPPLER_060
 #cmakedefine HAVE_POPPLER_030
 #cmakedefine HAVE_POPPLER_020
diff -urN tdegraphics.orig/tdefile-plugins/dependencies/poppler-tqt/ConfigureChecks.cmake tdegraphics/tdefile-plugins/dependencies/poppler-tqt/ConfigureChecks.cmake
--- tdegraphics.orig/tdefile-plugins/dependencies/poppler-tqt/ConfigureChecks.cmake	2019-02-19 23:03:51.960567176 +0300
+++ tdegraphics/tdefile-plugins/dependencies/poppler-tqt/ConfigureChecks.cmake	2019-02-19 23:06:54.509563813 +0300
@@ -34,3 +34,14 @@
     message( STATUS "Performing Test HAVE_POPPLER_060 - Failed" )
   endif( )
 endif( )
+
+if( NOT DEFINED HAVE_POPPLER_064 )
+  message( STATUS "Performing Test HAVE_POPPLER_064" )
+  if( NOT POPPLER_VERSION VERSION_LESS "0.64" )
+    set( HAVE_POPPLER_064 1 CACHE INTERNAL "" FORCE )
+    message( STATUS "Performing Test HAVE_POPPLER_064 - Success" )
+  else( )
+    set( HAVE_POPPLER_064 "" CACHE INTERNAL "" FORCE )
+    message( STATUS "Performing Test HAVE_POPPLER_064 - Failed" )
+  endif( )
+endif( )
diff -urN tdegraphics.orig/tdefile-plugins/dependencies/poppler-tqt/poppler-document.cc tdegraphics/tdefile-plugins/dependencies/poppler-tqt/poppler-document.cc
--- tdegraphics.orig/tdefile-plugins/dependencies/poppler-tqt/poppler-document.cc	2019-02-19 23:03:51.960567176 +0300
+++ tdegraphics/tdefile-plugins/dependencies/poppler-tqt/poppler-document.cc	2019-02-19 23:10:58.430559319 +0300
@@ -164,6 +164,8 @@
   TQString result;
   Object obj;
   GooString *s1;
+  CONST_064 GooString *s1;
+  GooString *s1;
   GBool isUnicode;
   Unicode u;
   int i;
@@ -320,7 +322,7 @@
   if ( !outline )
     return NULL;
 
-  GooList * items = outline->getItems();
+  CONST_064 GooList * items = outline->getItems();
   if ( !items || items->getLength() < 1 )
     return NULL;
 
diff -urN tdegraphics.orig/tdefile-plugins/dependencies/poppler-tqt/poppler-link.cc tdegraphics/tdefile-plugins/dependencies/poppler-tqt/poppler-link.cc
--- tdegraphics.orig/tdefile-plugins/dependencies/poppler-tqt/poppler-link.cc	2019-02-19 23:03:51.960567176 +0300
+++ tdegraphics/tdefile-plugins/dependencies/poppler-tqt/poppler-link.cc	2019-02-19 23:13:09.285556909 +0300
@@ -30,7 +30,7 @@
 	LinkDestination::LinkDestination(const LinkDestinationData &data)
 	{
 		bool deleteDest = false;
-		LinkDest *ld = data.ld;
+		CONST_064 LinkDest *ld = data.ld;
 		
 		if ( data.namedDest && !ld )
 		{
diff -urN tdegraphics.orig/tdefile-plugins/dependencies/poppler-tqt/poppler-private.cc tdegraphics/tdefile-plugins/dependencies/poppler-tqt/poppler-private.cc
--- tdegraphics.orig/tdefile-plugins/dependencies/poppler-tqt/poppler-private.cc	2019-02-19 23:03:51.960567176 +0300
+++ tdegraphics/tdefile-plugins/dependencies/poppler-tqt/poppler-private.cc	2019-02-19 23:22:36.329546462 +0300
@@ -31,7 +31,7 @@
 namespace Poppler {
 
 /* borrowed from kpdf */
-TQString unicodeToTQString(Unicode* u, int len)
+TQString unicodeToTQString(CONST_064 Unicode* u, int len)
 {
     TQString ret;
     ret.setLength(len);
@@ -41,7 +41,7 @@
     return ret;
 }
 
-TQString UnicodeParsedString(GooString *s1)
+TQString UnicodeParsedString(CONST_064 GooString *s1)
 {
     GBool isUnicode;
     int i;
@@ -86,7 +86,7 @@
 }
 
 
-void DocumentData::addTocChildren( TQDomDocument * docSyn, TQDomNode * parent, GooList * items )
+void DocumentData::addTocChildren( TQDomDocument * docSyn, TQDomNode * parent, CONST_064 GooList * items )
 {
     int numItems = items->getLength();
     for ( int i = 0; i < numItems; ++i )
@@ -96,7 +96,7 @@
 
         // 1. create element using outlineItem's title as tagName
         TQString name;
-        Unicode * uniChar = outlineItem->getTitle();
+        CONST_064 Unicode * uniChar = outlineItem->getTitle();
         int titleLength = outlineItem->getTitleLength();
         name = unicodeToTQString(uniChar, titleLength);
         if ( name.isEmpty() )
@@ -106,18 +106,18 @@
         parent->appendChild( item );
 
         // 2. find the page the link refers to
-        ::LinkAction * a = outlineItem->getAction();
+        CONST_064 ::LinkAction * a = outlineItem->getAction();
         if ( a && ( a->getKind() == actionGoTo || a->getKind() == actionGoToR ) )
         {
             // page number is contained/referenced in a LinkGoTo
-            LinkGoTo * g = static_cast< LinkGoTo * >( a );
-            LinkDest * destination = g->getDest();
+            CONST_064 LinkGoTo * g = static_cast< CONST_064 LinkGoTo * >( a );
+            CONST_064 LinkDest * destination = g->getDest();
             if ( !destination && g->getNamedDest() )
             {
                 // no 'destination' but an internal 'named reference'. we could
                 // get the destination for the page now, but it's VERY time consuming,
                 // so better storing the reference and provide the viewport on demand
-                GooString *s = g->getNamedDest();
+                CONST_064 GooString *s = g->getNamedDest();
                 TQChar *charArray = new TQChar[s->getLength()];
                 for (int i = 0; i < s->getLength(); ++i) charArray[i] = TQChar(s->getCString()[i]);
                     TQString aux(charArray, s->getLength());
@@ -131,14 +131,14 @@
                 }
                 if ( a->getKind() == actionGoToR )
                 {
-                    LinkGoToR * g2 = static_cast< LinkGoToR * >( a );
+                    CONST_064 LinkGoToR * g2 = static_cast< CONST_064 LinkGoToR * >( a );
                     item.setAttribute( "ExternalFileName", g2->getFileName()->getCString() );
                 }
             }
 
         // 3. recursively descend over children
         outlineItem->open();
-        GooList * children = outlineItem->getKids();
+        CONST_064 GooList * children = outlineItem->getKids();
         if ( children )
             addTocChildren( docSyn, &item, children );
     }
diff -urN tdegraphics.orig/tdefile-plugins/dependencies/poppler-tqt/poppler-private.h tdegraphics/tdefile-plugins/dependencies/poppler-tqt/poppler-private.h
--- tdegraphics.orig/tdefile-plugins/dependencies/poppler-tqt/poppler-private.h	2019-02-19 23:03:51.960567176 +0300
+++ tdegraphics/tdefile-plugins/dependencies/poppler-tqt/poppler-private.h	2019-02-19 23:27:18.167541270 +0300
@@ -34,24 +34,29 @@
 #else
 class SplashOutputDev;
 #endif
+#if defined(HAVE_POPPLER_064)
+#define CONST_064 const
+#else
+#define CONST_064
+#endif
 
 namespace Poppler {
     
 class DocumentData;
 
-TQString unicodeToTQString(Unicode* u, int len);
+TQString unicodeToTQString(CONST_064 Unicode* u, int len);
 
-TQString UnicodeParsedString(GooString *s1);
+TQString UnicodeParsedString(CONST_064 GooString *s1);
 
 GooString *TQStringToGooString(const TQString &s);
 
 class LinkDestinationData {
   public:
-     LinkDestinationData( LinkDest *l, GooString *nd, Poppler::DocumentData *pdfdoc ) : ld(l), namedDest(nd), doc(pdfdoc)
+     LinkDestinationData( CONST_064 LinkDest *l, GooString *nd, Poppler::DocumentData *pdfdoc ) : ld(l), namedDest(nd), doc(pdfdoc)
      {
      }
 	
-     LinkDest *ld;
+     CONST_064 LinkDest *ld;
      GooString *namedDest;
      Poppler::DocumentData *doc;
 };
@@ -84,7 +89,7 @@
         return m_outputDev;
     }
 
-    void addTocChildren( TQDomDocument * docSyn, TQDomNode * parent, GooList * items );
+    void addTocChildren( TQDomDocument * docSyn, TQDomNode * parent, CONST_064 GooList * items );
 
   class PDFDoc doc;
   bool locked;
